vis/optimizer.py

```diff
@@ -93,7 +93,8 @@ def _get_seed_input(self, seed_input):
             seed_input = np.expand_dims(seed_input, 0)
 
         # Only possible if channel idx is out of place.
-        if seed_input.shape != desired_shape:
+        if seed_input.shape[3] != desired_shape[3] and \
```

Comment:
better to use shape[-1] since the input could be n-dim tensor. looks good otherwise.

Reply:
fixed

Reply:
fixed. @raghakot do you have an idea how to fix the unit-test issues.